src/leveled_runner.erl

```diff
@@ -353,11 +353,16 @@ foldobjects_allkeys(SnapFun, Tag, FoldObjectsFun, sqn_order) ->
 
     Folder =
         fun() ->
-            {ok, LedgerSnapshot, JournalSnapshot, AfterFun} = SnapFun(),
-            {ok, JournalSQN} = leveled_inker:ink_getjournalsqn(JournalSnapshot),
+            {ok, LedgerSnapshot, JournalSnapshot, AfterFun} = 
+                case SnapFun() of
+                    {ok, LS, JS, AF} when is_pid(JS) ->
```

Comment:
This is really adding code for the sake of type checking... 
Not sure how to solve it... dependent type for `leveled_bookie:return_snapfun` possibly.

Reply:
As it is an object fold, it needs a snapshot of the journal - other folds don't.  Perhaps there needs to be two types of snap_fun()?  Though perhaps that just might squeeze the problem elsewhere.